engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -249,4 +265,31 @@ private static final class UnmappedActiveElementException extends RuntimeExcepti
               processInstanceKey, elementId));
     }
   }
+
+  /**
+   * Exception that can be thrown during the migration of a process instance, in case any of the
+   * mapping instructions of the command refer to a source and a target element with different
+   * element type, or different event type.
+   */
+  private static final class IncorrectMappingException extends RuntimeException {
```

Comment:
🔧 We can use a more expressive exception name, for example, `ElementTypeChangedException` or so.